53_inferance_pipeline

Sean1572 opened the pull request.

Sean1572 committed:
Added some experimental inferance pipeline

Needed this to test unseen burrowing owl dataset. Will help alot with getting data formatted correctly

Sean1572 committed:
Add testing code for loading in models for inferance

Sean1572 committed:
feat: add inferance pipeline

Sean1572 committed:
Block CSVs from commits

Sean1572 committed:
Move visualization packages to optional dependecy

Sean1572 committed:
Clean code (remove comments and remove run_spefific code from library)

Sean1572 commented:
@sumi1194 here is some code for the inference pipeline, let me know what you think

Sean1572 committed:
Clean up timm_model

Sean1572 committed:
72 inferance addons (#89)

* Analysis of predicted outputs

* Study on 2024 burrowing owl data predictions

Sean1572 committed:
Linted

Sean1572 committed:
Linted

Sean1572 committed:
Merge branch 'dev' into 72_inferance

Sean1572 committed:
Linted

kgarwoodsdzwa reviewed (changes requested):
do the jupyter notebook files need to be in this pr? seems like it should be separate- it's a lot to merge in one branch. the call_density, chunk_manual_data, and compare_model_outputs, label, and visualization

kgarwoodsdzwa commented on whoot_model_training/inference.py:
the docstring is a copy of the train.py, is not relevant to inferance. py. also- spelling check on inference- it's with an e instead of an a

kgarwoodsdzwa commented:
small idea, but maybe error handling for empty or corrupted soundfiles? i had 2 files wav files in my directory that were corrupted and one that was a wav but just 0kb, and until i moved them out of the folder id get an EOFError and a crash for running inference.

Sean1572 commented:
> do the jupyter notebook files need to be in this pr? seems like it should be separate- it's a lot to merge in one branch. the call_density, chunk_manual_data, and compare_model_outputs, label, and visualization

Some yes, some can be removed. Namely chunk_manual_data, compare_model_outputs, and label were debugging scripts and things I must have missed when prepping the PR

However, I think there is a good case for at least two of these to stay: call_density and visualization. I argue these are applications of the inference pipeline, examples of what can be done by this PR

Sean1572 committed:
fix: clean up notebooks

Sean1572 commented:
> small idea, but maybe error handling for empty or corrupted soundfiles? i had 2 files wav files in my directory that were corrupted and one that was a wav but just 0kb, and until i moved them out of the folder id get an EOFError and a crash for running inference.

Work with my san diego few shot model has some small error handling for training, not just on inference. I've implemented this at the extractor level. I think what this means is that there is a larger issue of corrupt file handling, I've made a new issue to address this in a diffrent pr #98

Sean1572 committed:
feat: clean inference.py and rename

kgarwoodsdzwa commented on whoot_model_training/whoot_model_training/models/timm_model.py:
```diff
@@ -20,7 +20,8 @@ class TimmInputs(ModelInput):
     """
-    def __init__(self, labels, waveform=None, spectrogram=None):
+
+    def __init__(self, labels, spectrogram=None):
```
why was the waveform=None removed? I had this back in order to get training to run on this branch. I had to retrain my panda model on owlorbit since it was located on cheetahchin and i needed a model to test the inference script, so when I used this branch to also train the model, I got an error here that waveform was not defined, so I had to add back waveform=None in order to get it to train.

kgarwoodsdzwa commented on whoot_model_training/inference.py:
```diff
@@ -0,0 +1,102 @@
+model type!)
+
+Usage:
+    $ python train.py /path/to/config.yml --model_name /path/to/model/parent/dir/
```
inference.py

Sean1572 committed:
Fix: return waveform in TimmInput

Sean1572 committed:
fix: Updated docs for inference

kgarwoodsdzwa commented:
for visualizations.ipynb, how did you get the buowset_preds_96545a6b81b24794962d464e8fcbdc1d.csv? Is it generated from the inference.py script? I don't see one output for when i ran inference.py. did you use a different script to generate that csv from the results?

kgarwoodsdzwa commented on whoot_model_training/inference.py:
```diff
@@ -0,0 +1,101 @@
+    """
+    # Extract a new dataset
+    unlabel_audio_path = "/mnt/restorage/Audiomoth/Raw sound files/2024/RGCB/"
+    run_name = f"buowset1.1_{model_name}_ATTEMPT_TO_STUDY_NEW_DATA"
```
I had to remove the {model_name} from this line in order for it to export the pkl at the end. when it was there it would crash at the end from this error: 
  File "/home/katie/whoot/whoot_model_training/panda_inference.py", line 86, in test
    with open(run_name + ".pkl", mode="wb") as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'pandas_multiclass_1.0_/home/katie/whoot/whoot_model_training/model_checkpoints/pandas_multiclass_1.0_efficientnet_b1_11_12_2025_16:55:31/checkpoint-775_ATTEMPT_TO_STUDY_NEW_DATA.pkl'

Sean1572 committed:
Untrack notebooks

Sean1572 committed:
Block notebooks in the future from being tracked

Sean1572 committed:
Fix missing path bug

Sean1572 committed:
Linting

kgarwoodsdzwa commented:
im thinking i will just merge it, i tested the updates and it works. I just still did have to add the class_list = [0,1,2] line in raw_data_extractor, because otherwise as-is it will valueerror out for needing to provide num_classes, names, or names_file. but I don't know if that's a huge deal, might just be my dataset specifically

kgarwoodsdzwa committed:
removed duplicated comet_workspace param

kgarwoodsdzwa committed:
typo inferance -> inference

kgarwoodsdzwa committed:
remove the labels value because it does not correlate to preds

the labels value would imply that each sample has the same label,
being the first class value. despite the prediction values
giving the actual logit predictions which do not
reflect that. removing these values removes confusion as the pred
values are what's more important anyways

kgarwoodsdzwa commented:
@Sean1572 im good to merge- but if you could review my most recent commit- i dont know if that's the best way to do it or if we should just solve the actual problem (the labels showing the incorrect values in the output). but we only really need the pred logits anyways. this is in regards to the labels dictionary value in the result only always showing [1, 0, 0], [1, 0, 0], [1, 0, 0] despite the preds showing the actual predictions of [-7.25150919e+00,  7.40287304e+00, -4.27632570e+00],
       [-8.90246391e+00,  6.94049358e+00, -3.94192100e+00],
       [ 4.47849369e+00, -5.96983910e+00,  5.62733793e+00]

kgarwoodsdzwa committed:
added comment to explain label delete line

should fix this later, but for now it gets rid of confusing output
that doesn't align labels to the predictions
